cmd/kubelet/app/server.go

```diff
@@ -560,8 +560,30 @@ func SimpleKubelet(client *client.Client,
 // Eventually, #2 will be replaced with instances of #3
 func RunKubelet(kcfg *KubeletConfig, builder KubeletBuilder) error {
 	kcfg.Hostname = nodeutil.GetHostname(kcfg.HostnameOverride)
```

Comment:
I don't think I am breaking hostname_override, except maybe on AWS, where if it is broken I want to break it :-)

The logic is this: we build Hostname, just as before: hostname from the OS, but can be overridden by hostname_override.  If we have a CloudProvider, we ask it to determine the node-name, but all the implementations apart from AWS simply echo back the maybe-overridden hostname.

That is why we pass hostname in to CurrentNodeName.  It is admittedly a weird argument to pass in - it feels a little out of place.  Maybe it would be cleaner if we didn't pass in anything, but only changed the nodeName if CurrentNodeName returned non-empty?  But then that makes the function even more specialized.  I think post V1 we will want a function which returns the CurrentNode() (see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L721-L735)

hostname (& hostname_override) do determine the SSL self-signed cert subject, so I think we have to keep this.

I do agree that we will likely end up with a nodename_override (or just nodename?) but I'm hoping that maybe we can avoid it - that it will only be needed on bare-metal, and on bare-metal the hostname is the logical identifier for nodes.

This is possibly optimistic, but I'd rather avoid introducing another flag until we have someone that needs it!

If you'd prefer, I can create nodename_override, but I don't think anyone will have to use it today.